### PR TITLE
Ports air traffic control from Paradise

### PR DESCRIPTION
**Pull request by ghost**
ported by steamp0rt from paradise, which was ported by tigercat2000 from polaris, where Neerti ported it from Vore, where it was coded by Arokha
:cl: steamp0rt, tigercat2000, Neerti, Arokha
add: Adds air traffic control! It's random space traffic control chatter during the game, over the Common channel.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
How's this look on lore?

File: sunset/code/modules/busy_space/air_traffic.dm [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+/datum/lore/atc_controller/proc/atc_process()
+	if(world.time >= next_message)
+		if(squelched)
+			next_message = world.time + backoff_delay
+		else
+			next_message = world.time + rand(delay_min,delay_max)
+			random_convo()
+
+	addtimer(CALLBACK(src, .proc/atc_process), 1 MINUTES)
```
Comment (alexkar598) [inline sunset/code/modules/busy_space/air_traffic.dm:27]: Why the unholy mix of timers and world.time. Use timer delays instead.

File: code/modules/security_levels/security_levels.dm [hunk 2]
```diff
@@ -81,6 +81,7 @@ GLOBAL_VAR_INIT(security_level, SEC_LEVEL_GREEN)
 					playsound(D, 'sound/machines/boltsup.ogg', 50, TRUE)
 		SSblackbox.record_feedback("tally", "security_level_changes", 1, get_security_level())
 		SSnightshift.check_nightshift()
+		sunset_get_atc_to_fuck_off(level) // sunset
```
Comment (alexkar598) [inline code/modules/security_levels/security_levels.dm]: Name is confusing,you would expect get_atc_to_fuck_off to actually mute ATC yet this runs a check

Perhaps sunset_check_atc_security_level?

File: sunset/code/modules/busy_space/air_traffic.dm [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	var/full_closure
+
+	switch(chatter_type)
+		if("wrong_freq")
+			callname = pick(wrong_atc_names)
+			full_request = "[callname], this is [combined_name] on a [mission] [pick(mission_noun)] to [destname], [pick(request_verb)] [request]."
+			full_response = "[combined_name], this is [GLOB.using_map.station_short] TraCon, wrong frequency. Switch to [rand(700,999)].[rand(1,9)]."
+			full_closure = "[GLOB.using_map.station_short] TraCon, copy, apologies."
+		if("wrong_lang")
+			//Can't implement this until autosay has language support
```
Comment (alexkar598) [inline sunset/code/modules/busy_space/air_traffic.dm]: If it’s not supported just don’t add it. You can come back at it later when it’s supported

File: sunset/code/modules/map_fluff/maps.dm [hunk 4]
```diff
@@ -0,0 +1,13 @@
+
+GLOBAL_DATUM_INIT(using_map, /datum/map/sunset, new) // too lazy to make a dynamic system for this
```
Comment (alexkar598) [inline sunset/code/modules/map_fluff/maps.dm]: Just load it via JSON info to the map config and access it with SSmapping.config instead of a hacky system

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: sunset/code/modules/busy_space/air_traffic.dm (hunk 1 above)
Comment (ghost) [reply sunset/code/modules/busy_space/air_traffic.dm:27]: god no i don't want to refactor half of this

File: sunset/code/modules/map_fluff/maps.dm (hunk 4 above)
Comment (ghost) [reply sunset/code/modules/map_fluff/maps.dm]: hm good idea

Comment (ghost) [reply sunset/code/modules/map_fluff/maps.dm]: Done

**Pull request merged**